File: Mobile/AppForCustomer/detect_care_app/lib/features/subscription/models/subscription.dart
class Subscription {
  final String id;
  final String userId;
  final String planId;
  final DateTime startedAt;
  final DateTime endsAt;
  final String status; // active, paused, cancelled
  final DateTime createdAt;
  final DateTime updatedAt;

  Subscription({
    required this.id,
    required this.userId,
    required this.planId,
    required this.startedAt,
    required this.endsAt,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Subscription.fromJson(Map<String, dynamic> json) {
    return Subscription(
      id: json['subscription_id'] ?? '',
      userId: json['user_id'] ?? '',
      planId: json['code'] ?? '',
      startedAt: DateTime.parse(
        json['started_at'] ?? DateTime.now().toIso8601String(),
      ),
      endsAt: DateTime.parse(
        json['current_period_end'] ?? DateTime.now().toIso8601String(),
      ),
      status: json['status'] ?? 'active',
      createdAt: DateTime.parse(
        json['started_at'] ?? DateTime.now().toIso8601String(),
      ),
      updatedAt: DateTime.parse(
        json['started_at'] ?? DateTime.now().toIso8601String(),
      ),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'plan_id': planId,
      'started_at': startedAt.toIso8601String(),
      'ends_at': endsAt.toIso8601String(),
      'status': status,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}

File: FE/vision_ai_admin_dashboard/src/types/subscription.ts
import { User } from './user';

export interface Plan {
  code: string;
  name: string;
  price: string;
  camera_quota: number;
  retention_days: number;
  caregiver_seats: number;
  sites: number;
  major_updates_months: number;
  created_at: string;
  storage_size: string;
  is_recommended: boolean;
  tier: number;
  currency: string;
  status: string;
  is_current: null;
  version: null;
  effective_from: null;
  effective_to: null;
}

export interface Subscription {
  subscription_id: string;
  user_id: string;
  plan_code: string;
  plan_id: string | null;
  status: 'active' | 'inactive' | 'cancelled' | 'expired' | string;
  billing_period: string;
  started_at: string;
  current_period_start: string;
  current_period_end: string | null;
  trial_end_at: string | null;
  canceled_at: string | null;
  ended_at: string | null;
  auto_renew: boolean;
  cancel_at_period_end: boolean;
  extra_camera_quota: number;
  extra_caregiver_seats: number;
  extra_sites: number;
  extra_storage_gb: number;
  notes: string | null;
  last_payment_at: string | null;
  version: null;
  plans: Plan | null;
  offer_start_date?: string | null;
  offer_end_date?: string | null;
  renewal_attempt_count?: number;
  next_renew_attempt_at?: string | null;
  dunning_stage?: string | null;
  last_renewal_error?: string | null;
  user?: User;
}

export type SubscriptionsListApiResponse = {
  success: boolean;
  data: Subscription[];
  pagination: {
    total: number;
    page: number;
    limit: number;
    totalPages?: number;
  };
  message?: string;
  timestamp?: string;
};

export interface Payment {
  payment_id: string;
  subscription_id: string;
  customer_id: number;
  amount: number;
  currency: string;
  status: 'pending' | 'completed' | 'failed' | 'refunded';
  payment_method: string;
  transaction_id?: string;
  created_at: string;
  updated_at: string;
}
